scripts/globals/moghouse.lua

```diff
@@ -239,3 +268,76 @@ xi.moghouse.addMogLockerExpiryTime = function(player, numBronze)
 
     return true
 end
+
+xi.moghouse.exitJobChange = function(player, prevZone)
+    if player:getCharVar('[Moghouse]Exit_Pending') == 1 and prevZone == player:getZoneID() then
+        player:setVolatileCharVar('[Moghouse]Exit_Pending', 0)
+        if xi.settings.map.MH_EXIT_HOMEPOINT then
+            if player:getCharVar('[Moghouse]Exit_Job_Change') == 1 and not player:isCurrentHomepoint() then
+                player:timer(100, function(playerArg)
+                    playerArg:startEvent(30004, 1)
+                end)
+            end
+        end
+    else
+        player:setVolatileCharVar('[Moghouse]Exit_Pending', 0)
+        player:setVolatileCharVar('[Moghouse]Exit_Job_Change', 0)
+    end
+end
+
+xi.moghouse.exitJobChangeFinish = function(player, csid, option)
+    if xi.settings.map.MH_EXIT_HOMEPOINT then
+        if csid == 30004 and option == 0 then
+            player:setHomePoint()
+        elseif csid == 30004 then
+            player:setVolatileCharVar('[Moghouse]Exit_Job_Change', 0)
+        end
+    end
+end
+
+xi.moghouse.nationrRentARoom = function(player, npc, zoneTable, regionId)
+    local rentARoomRegion = player:getCharVar("[Moghouse]Rent-A-Room")
+    if player:getNation() == zoneTable.nation then
+        if player:getCharVar("[Moghouse]GuardSpeach") == 0 then -- First time talking to guard.
+            player:setCharVar("[Moghouse]GuardSpeach", 1)
+            player:startEvent(zoneTable.csBase)
+        elseif rentARoomRegion ~= regionId and xi.settings.map.ERA_RENT_A_ROOM then -- Needs to move room.
+            player:startEvent(zoneTable.moveRoom, xi.moghouse.rentARoomRegionBits[rentARoomRegion])
+        else
+            player:startEvent(zoneTable.repeatCS) -- Second time talking to guard with room in city.
+        end
+    else
+        utils.ternary(rentARoomRegion ~= regionId, player:startEvent(zoneTable.needRent), player:startEvent(zoneTable.hasRent)) -- Needs Registration | Already Registered
```

Comment:
nit: I'd prefer this as not a ternary since you're not doing an assignment? Don't care that strongly

Reply:
Changed.